tool/tsh/gcp.go

```diff
@@ -126,9 +129,35 @@ func (a *gcpApp) Close() error {
 	return trace.NewAggregate(errs...)
 }
 
+func (a *gcpApp) getGcloudConfigPath() string {
+	return path.Join(profile.FullProfilePath(a.cf.HomePath), "gcp", a.app.ClusterName, a.app.Name, "gcloud")
+}
+
+var projectIDRegexp = regexp.MustCompile(`.*@(?P<projectID>.*)\.iam\.gserviceaccount\.com`)
+
+func projectIDFromServiceAccountName(serviceAccount string) (string, error) {
+	match := projectIDRegexp.FindStringSubmatch(serviceAccount)
+	if match == nil {
+		return "", trace.BadParameter("cannot extract project ID from service account %q", serviceAccount)
+	}
+
+	for i, name := range projectIDRegexp.SubexpNames() {
+		if name == "projectID" {
+			return match[i], nil
+		}
+	}
+
+	return "", trace.BadParameter("cannot extract project ID from service account %q", serviceAccount)
```

Comment:
Any regexp instead of  string split ? Personally for me split is more readable that reading regxp expression

Reply:
Yeah, I was on the fence about this. I dislike regexps but this looked like a good use case. I can easily reimplement this using split, though.

Reply:
Current version LGTM so this is only my personal bypass agains regex so leaving it up to you.

Reply:
I gave it a go, and I like the non-regexp version better. It is easier to reason about and leads to much nicer error messages, which is important for validation.